altair/vegalite/v5/schema/channels.py

```diff
@@ -24641,45 +24487,45 @@ def field(
+        clamp: Optional[Map | bool | Parameter | SchemaBase] = Undefined,
+        constant: Optional[Map | float | Parameter | SchemaBase] = Undefined,
         domain: Optional[
-            dict
+            Map
             | Parameter
             | SchemaBase
             | Literal["unaggregated"]
-            | Sequence[str | bool | dict | None | float | Parameter | SchemaBase]
+            | Sequence[Map | str | bool | None | float | Parameter | SchemaBase]
         ] = Undefined,
-        domainMax: Optional[dict | float | Parameter | SchemaBase] = Undefined,
-        domainMid: Optional[dict | float | Parameter | SchemaBase] = Undefined,
-        domainMin: Optional[dict | float | Parameter | SchemaBase] = Undefined,
-        domainRaw: Optional[dict | Parameter | SchemaBase] = Undefined,
-        exponent: Optional[dict | float | Parameter | SchemaBase] = Undefined,
+        domainMax: Optional[Map | float | Parameter | SchemaBase] = Undefined,
+        domainMid: Optional[Map | float | Parameter | SchemaBase] = Undefined,
+        domainMin: Optional[Map | float | Parameter | SchemaBase] = Undefined,
+        domainRaw: Optional[Map | Parameter | SchemaBase] = Undefined,
+        exponent: Optional[Map | float | Parameter | SchemaBase] = Undefined,
         interpolate: Optional[
-            dict | Parameter | SchemaBase | ScaleInterpolateEnum_T
+            Map | Parameter | SchemaBase | ScaleInterpolateEnum_T
         ] = Undefined,
         nice: Optional[
-            bool | dict | float | Parameter | SchemaBase | TimeInterval_T
+            Map | bool | float | Parameter | SchemaBase | TimeInterval_T
         ] = Undefined,
-        padding: Optional[dict | float | Parameter | SchemaBase] = Undefined,
-        paddingInner: Optional[dict | float | Parameter | SchemaBase] = Undefined,
-        paddingOuter: Optional[dict | float | Parameter | SchemaBase] = Undefined,
+        padding: Optional[Map | float | Parameter | SchemaBase] = Undefined,
+        paddingInner: Optional[Map | float | Parameter | SchemaBase] = Undefined,
+        paddingOuter: Optional[Map | float | Parameter | SchemaBase] = Undefined,
         range: Optional[
-            dict
+            Map
             | SchemaBase
-            | Sequence[str | dict | float | Parameter | SchemaBase | Sequence[float]]
+            | Sequence[Map | str | float | Parameter | SchemaBase | Sequence[float]]
             | RangeEnum_T
         ] = Undefined,
-        rangeMax: Optional[str | dict | float | Parameter | SchemaBase] = Undefined,
-        rangeMin: Optional[str | dict | float | Parameter | SchemaBase] = Undefined,
-        reverse: Optional[bool | dict | Parameter | SchemaBase] = Undefined,
-        round: Optional[bool | dict | Parameter | SchemaBase] = Undefined,
-        scheme: Optional[dict | Parameter | SchemaBase | ColorScheme_T] = Undefined,
+        rangeMax: Optional[Map | str | float | Parameter | SchemaBase] = Undefined,
```

Comment:
Thanks @dangotbanned!

Any chance we could have `Map` later on before `Parameter or SchemaBase`? Maybe this was also not done consistently before, there is some code in `tools/schemapi/utils.py:488`.

I'm bringing it up in this PR as `Map` is now first in some cases where it was not before which is a slight deterioration in UX.

Reply:
Yeah should be able to sort that out, so `Map` after all lowercase?

Reply:
I think that would make sense, maybe even after Parameter? To me, it seems like the basic types (`str`, `float`, ...) and `Parameter` are what users are interested in.

Reply:
I can get this without needing to special-case anything:

<details>
<summary>Huge output diff screenshot</summary>

![image](https://github.com/user-attachments/assets/14296487-3073-46e4-8d50-95c01ab61140)

</details>

Essentially moves `Map` into the aliases group, and since it is the shortest - it appears first

<details>
<summary>Required changes diff</summary>

![image](https://github.com/user-attachments/assets/44d287ad-bf8d-4c74-a795-7d54bd5b48a6)


</details>

Would that be a reasonable fix @binste?

Trying it visually here - since the actual diff this produces is gigantic